### PR TITLE
Snowflake permission note

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?

See internal chat for context:  https://dbt-labs.slack.com/archives/C02NCQ9483C/p1732724200239919?thread_ts=1732650446.311849&cid=C02NCQ9483C

## Checklist
- [ ] I have reviewed the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] The topic I'm writing about is for specific dbt version(s) and I have versioned it according to the [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and/or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content) guidelines.
- [ ] I have added checklist item(s) to this list for anything anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."
- [ ] The content in this PR requires a dbt release note, so I added one to the [release notes page](https://docs.getdbt.com/docs/dbt-versions/dbt-cloud-release-notes).

---
🚀 Deployment available! Here are the direct links to the updated files:

- https://docs-getdbt-com-git-snowflake-permissions-dbt-labs.vercel.app/docs/cloud/connect-data-platform/connect-snowflake

### CODE REVIEW

File: website/docs/docs/cloud/connect-data-platform/connect-snowflake.md [hunk 1]
```diff
@@ -5,6 +5,14 @@ description: "Configure Snowflake connection."
 sidebar_label: "Connect Snowflake"
 ---
 
+:::note
+
+A crucial part of working with dbt atop Snowflake is ensuring that users in development environments and/or service accounts in deployment to production environments have the correct permissions to take actions on Snowflake.
+
+The roles and associated permissions in Snowflake can be customized to fit your company's requirements and fine-tune access to database objects in your account. See [Snowflake permissions](/reference/database-permissions/snowflake-permissions) for more information about customizing roles in Snowflake.
```
Comment (matthewshaver) [inline website/docs/docs/cloud/connect-data-platform/connect-snowflake.md]:
```suggestion
You can customize roles and associated permissions in Snowflake to fit your company's requirements and fine-tune access to database objects in your account. See [Snowflake permissions](/reference/database-permissions/snowflake-permissions) for more information about customizing roles in Snowflake.
```

File: website/docs/docs/cloud/connect-data-platform/connect-snowflake.md [hunk 2]
```diff
@@ -5,6 +5,14 @@ description: "Configure Snowflake connection."
 sidebar_label: "Connect Snowflake"
 ---
 
+:::note
+
+A crucial part of working with dbt atop Snowflake is ensuring that users in development environments and/or service accounts in deployment to production environments have the correct permissions to take actions on Snowflake.
```
Comment (mirnawong1) [inline website/docs/docs/cloud/connect-data-platform/connect-snowflake.md]: i'm not sure if we need this intro here
```suggestion
dbt Cloud works with Snowflake by inheriting permissions directly from your Snowflake data platform configuration. This ensures you have full control over what dbt Cloud can access. You can customize roles and permissions in Snowflake to fit your company's requirements and manage access to database objects. 
```

File: website/docs/docs/cloud/connect-data-platform/connect-snowflake.md [hunk 3]
```diff
@@ -5,6 +5,14 @@ description: "Configure Snowflake connection."
...
+
+You can customize roles and associated permissions in Snowflake to fit your company's requirements and fine-tune access to database objects in your account. See [Snowflake permissions](/reference/database-permissions/snowflake-permissions) for more information about customizing roles in Snowflake.
```
Comment (mirnawong1) [inline website/docs/docs/cloud/connect-data-platform/connect-snowflake.md]:
```suggestion
Refer to [Snowflake permissions](/reference/database-permissions/snowflake-permissions) for more information about customizing roles in Snowflake.
```

**Review by mirnawong1 — COMMENTED** [review]
hey @matthewshaver , left some suggestions here!

File: website/docs/docs/cloud/connect-data-platform/connect-snowflake.md (hunk 2 above)
Comment (matthewshaver) [reply website/docs/docs/cloud/connect-data-platform/connect-snowflake.md]: That's the original not deleted below. I'm ok with removing it, but we should be careful with this language. We don't want to inadvertently indicate that Snowflake user permissions can somehow be assigned to users in dbt Cloud itself (for their permissions over the dbt account). How about:

```suggestion
dbt Cloud connections and credentials inherit the permissions of the accounts configured. You can customize roles and associated permissions in Snowflake to fit your company's requirements and fine-tune access to database objects in your account. See [Snowflake permissions](/reference/database-permissions/snowflake-permissions) for more information about customizing roles in Snowflake.
```

**Review by mirnawong1 — APPROVED** [review]
oops forgot to approve

**Pull request merged**